{
	"document_title": "Security update for python3",
	"document_type": "SUSE Patch",
	"document_publisher": {
		"type": "Vendor",
		"contact_details": "security@suse.de",
		"issuing_authority": "SUSE Security Team"
	},
	"document_tracking": {
		"identification": {
			"id": "SUSE-SU-2024:3351-1"
		},
		"status": "Final",
		"version": "1",
		"revision_history": {
			"revision": {
				"number": "1",
				"date": "2024-09-20T07:58:24Z",
				"description": "current"
			}
		},
		"initial_release_date": "2024-09-20T07:58:24Z",
		"current_release_date": "2024-09-20T07:58:24Z",
		"generator": {
			"engine": "cve-database/bin/generate-cvrf.pl",
			"date": "2017-02-24T01:00:00Z"
		}
	},
	"document_notes": [
		{
			"text": "Security update for python3",
			"title": "Topic",
			"type": "Summary"
		},
		{
			"text": "This update for python3 fixes the following issues:\n\n- CVE-2024-5642: buffer overread when NPN is used and invalid values are sent to the OpenSSL API. (bsc#1227233)\n",
			"title": "Details",
			"type": "General"
		},
		{
			"text": "The CVRF data is provided by SUSE under the Creative Commons License 4.0 with Attribution (CC-BY-4.0).",
			"title": "Terms of Use",
			"type": "Legal Disclaimer"
		},
		{
			"text": "SUSE-2024-3351,SUSE-SUSE-MicroOS-5.1-2024-3351",
			"title": "Patchnames",
			"type": "Details"
		}
	],
	"documentdistribution": "Copyright SUSE LLC under the Creative Commons License 4.0 with Attribution (CC-BY-4.0)",
	"document_references": [
		{
			"type": "Self",
			"url": "https://www.suse.com/support/update/announcement/2024/suse-su-20243351-1/",
			"description": "Link for SUSE-SU-2024:3351-1"
		},
		{
			"type": "Self",
			"url": "https://lists.suse.com/pipermail/sle-security-updates/2024-September/019481.html",
			"description": "E-Mail link for SUSE-SU-2024:3351-1"
		},
		{
			"type": "Self",
			"url": "https://www.suse.com/support/security/rating/",
			"description": "SUSE Security Ratings"
		},
		{
			"type": "Self",
			"url": "https://bugzilla.suse.com/1227233",
			"description": "SUSE Bug 1227233"
		},
		{
			"type": "Self",
			"url": "https://www.suse.com/security/cve/CVE-2024-5642/",
			"description": "SUSE CVE CVE-2024-5642 page"
		}
	],
	"product_tree": {
		"branch": [
			{
				"type": "Product Family",
				"name": "SUSE Linux Enterprise Micro 5.1",
				"branch": [
					{
						"type": "Product Name",
						"name": "SUSE Linux Enterprise Micro 5.1",
						"full_product_name": {
							"text": "SUSE Linux Enterprise Micro 5.1",
							"product_id": "SUSE Linux Enterprise Micro 5.1",
							"cpe": "cpe:/o:suse:suse-microos:5.1"
						}
					}
				],
				"full_product_name": {}
			},
			{
				"type": "Product Version",
				"name": "libpython3_6m1_0-3.6.15-150000.3.158.1",
				"full_product_name": {
					"text": "libpython3_6m1_0-3.6.15-150000.3.158.1",
					"product_id": "libpython3_6m1_0-3.6.15-150000.3.158.1"
				}
			},
			{
				"type": "Product Version",
				"name": "libpython3_6m1_0-32bit-3.6.15-150000.3.158.1",
				"full_product_name": {
					"text": "libpython3_6m1_0-32bit-3.6.15-150000.3.158.1",
					"product_id": "libpython3_6m1_0-32bit-3.6.15-150000.3.158.1"
				}
			},
			{
				"type": "Product Version",
				"name": "libpython3_6m1_0-64bit-3.6.15-150000.3.158.1",
				"full_product_name": {
					"text": "libpython3_6m1_0-64bit-3.6.15-150000.3.158.1",
					"product_id": "libpython3_6m1_0-64bit-3.6.15-150000.3.158.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python3-3.6.15-150000.3.158.1",
				"full_product_name": {
					"text": "python3-3.6.15-150000.3.158.1",
					"product_id": "python3-3.6.15-150000.3.158.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python3-base-3.6.15-150000.3.158.1",
				"full_product_name": {
					"text": "python3-base-3.6.15-150000.3.158.1",
					"product_id": "python3-base-3.6.15-150000.3.158.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python3-curses-3.6.15-150000.3.158.1",
				"full_product_name": {
					"text": "python3-curses-3.6.15-150000.3.158.1",
					"product_id": "python3-curses-3.6.15-150000.3.158.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python3-dbm-3.6.15-150000.3.158.1",
				"full_product_name": {
					"text": "python3-dbm-3.6.15-150000.3.158.1",
					"product_id": "python3-dbm-3.6.15-150000.3.158.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python3-devel-3.6.15-150000.3.158.1",
				"full_product_name": {
					"text": "python3-devel-3.6.15-150000.3.158.1",
					"product_id": "python3-devel-3.6.15-150000.3.158.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python3-doc-3.6.15-150000.3.158.1",
				"full_product_name": {
					"text": "python3-doc-3.6.15-150000.3.158.1",
					"product_id": "python3-doc-3.6.15-150000.3.158.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python3-doc-devhelp-3.6.15-150000.3.158.1",
				"full_product_name": {
					"text": "python3-doc-devhelp-3.6.15-150000.3.158.1",
					"product_id": "python3-doc-devhelp-3.6.15-150000.3.158.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python3-idle-3.6.15-150000.3.158.1",
				"full_product_name": {
					"text": "python3-idle-3.6.15-150000.3.158.1",
					"product_id": "python3-idle-3.6.15-150000.3.158.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python3-testsuite-3.6.15-150000.3.158.1",
				"full_product_name": {
					"text": "python3-testsuite-3.6.15-150000.3.158.1",
					"product_id": "python3-testsuite-3.6.15-150000.3.158.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python3-tk-3.6.15-150000.3.158.1",
				"full_product_name": {
					"text": "python3-tk-3.6.15-150000.3.158.1",
					"product_id": "python3-tk-3.6.15-150000.3.158.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python3-tools-3.6.15-150000.3.158.1",
				"full_product_name": {
					"text": "python3-tools-3.6.15-150000.3.158.1",
					"product_id": "python3-tools-3.6.15-150000.3.158.1"
				}
			}
		],
		"relationship": [
			{
				"product_reference": "libpython3_6m1_0-3.6.15-150000.3.158.1",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "SUSE Linux Enterprise Micro 5.1"
			},
			{
				"product_reference": "python3-3.6.15-150000.3.158.1",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "SUSE Linux Enterprise Micro 5.1"
			},
			{
				"product_reference": "python3-base-3.6.15-150000.3.158.1",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "SUSE Linux Enterprise Micro 5.1"
			}
		]
	},
	"vulnerability": [
		{
			"notes": [
				{
					"text": "CPython 3.9 and earlier doesn't disallow configuring an empty list (\"[]\") for SSLContext.set_npn_protocols() which is an invalid value for the underlying OpenSSL API. This results in a buffer over-read when NPN is used (see CVE-2024-5535 for OpenSSL). This vulnerability is of low severity due to NPN being not widely used and specifying an empty list likely being uncommon in-practice (typically a protocol name would be configured).",
					"title": "Vulnerability Description",
					"type": "General"
				}
			],
			"cve": "CVE-2024-5642",
			"product_statuses": [
				{
					"type": "Fixed",
					"product_id": [
						"SUSE Linux Enterprise Micro 5.1:libpython3_6m1_0-3.6.15-150000.3.158.1",
						"SUSE Linux Enterprise Micro 5.1:python3-3.6.15-150000.3.158.1",
						"SUSE Linux Enterprise Micro 5.1:python3-base-3.6.15-150000.3.158.1"
					]
				}
			],
			"threats": [
				{
					"type": "Impact",
					"description": "moderate"
				}
			],
			"cvss_score_sets": {
				"score_set_v_2": {},
				"score_set_v_3": {
					"base_score_v_3": "4.8",
					"vector_v_3": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:L/I:N/A:L"
				},
				"scoreset": {}
			},
			"remediations": [
				{
					"type": "Vendor Fix",
					"description": {
						"text": "To install this SUSE Security Update use the SUSE recommended installation methods like YaST online_update or \"zypper patch\".\n",
						"lang": "en"
					},
					"url": "https://www.suse.com/support/update/announcement/2024/suse-su-20243351-1/"
				}
			],
			"references": [
				{
					"url": "https://www.suse.com/security/cve/CVE-2024-5642.html",
					"description": "CVE-2024-5642"
				},
				{
					"url": "https://bugzilla.suse.com/1227233",
					"description": "SUSE Bug 1227233"
				}
			]
		}
	]
}
